apps/src/studio/studio.js

```diff
@@ -2643,12 +2702,23 @@ Studio.allGoalsVisited = function() {
     var goal = Studio.spriteGoals_[i];
     if (!goal.finished) {
       if (protagonistSprite) {
+        var wasGoalFinished = goal.finished;
+
         goal.finished = spriteAtGoal(protagonistSprite, goal);
+
+        // If goal was just finished, then call the "when actor touches anything handler"
+        if (! wasGoalFinished && goal.finished) {
+          var allowQueueExtension = false;
+          var prefix = 'whenSpriteCollided-' + Studio.protagonistSpriteIndex + '-';
+          callHandler(prefix + 'anything', allowQueueExtension);
+        }
+
```

Comment:
Nit: No space between ! and wasGoalFinished.
It is most likely safe, but I'm not 100% sure. We could get into trouble if there are existing levels that have both goals and collide with anything events.
Do we need to have score involved? It sounds like you're doing something like this level maybe? http://studio.code.org/s/playlab/stage/1/puzzle/6